### ISSUE TITLE
No CredentialsProvider on clone

### ISSUE THREAD

**Issue by hughstevenson** [oscerd/camel-git#1]
Hi oscerd

Started using the camel-git component and ran into an authentication issue when trying to clone from BitBucket. 

from("timer://cloneRepo?fixedRate=true&daemon=false&period=500s")
      .to("git://${gitLocalRepo}?remotePath=${remotePath}&operation=clone")

Looks like GitProducer.doClone() does not supply a CredentialsProvider object in the request. Get the below exception.
https://bitbucket.org/******.git: Authentication is required but no CredentialsProvider has been registered.

I've been able to clone a git repo that does not require authentication.

I can update a do pull request if you could let me know where the parent project is ?

<parent>
    <groupId>org.apache.camel</groupId>
    <artifactId>components</artifactId>
    <version>2.16-SNAPSHOT</version>
  </parent>

Regards
Hugh

**Comment by oscerd** [comment]
Hi Hugh,

Thank you.

The parent project is Apache Camel: https://github.com/apache/camel/tree/master/components/camel-git

In Camel there is the latest version of the component.

You can find all the informations about the component in the documentation:
http://camel.apache.org/git.html

As you can see you can specify username/password in the latest version.

Sorry if I don't specify the project was integrated in Camel. I will update the README.md in this repo accordingly :-)

Feel free to open an issue on the Camel JIRA if you find bugs, you want to introduce improvements and so on :-)

The JIRA is here: https://issues.apache.org/jira/browse/CAMEL

**Comment by oscerd** [comment]
You're right. We currently miss the credentials support in clone operation. I've logged a JIRA for this and I will fix it.

https://issues.apache.org/jira/browse/CAMEL-9386
